pkg/cacheutil/redis_client.go

```diff
@@ -161,39 +176,43 @@ func NewRedisClientWithConfig(logger log.Logger, name string, config RedisClient
 		return nil, err
 	}
 
-	opts := &redis.Options{
-		Addr:         config.Addr,
-		Username:     config.Username,
-		Password:     config.Password,
-		DB:           config.DB,
-		DialTimeout:  config.DialTimeout,
-		ReadTimeout:  config.ReadTimeout,
-		WriteTimeout: config.WriteTimeout,
-		MinIdleConns: config.MinIdleConns,
-		MaxConnAge:   config.MaxConnAge,
-		IdleTimeout:  config.IdleTimeout,
-	}
-
+	var tlsConfig *tls.Config
 	if config.TLSEnabled {
-		tlsConfig := config.TLSConfig
+		userTLSConfig := config.TLSConfig
 
-		tlsClientConfig, err := thanos_tls.NewClientConfig(logger, tlsConfig.CertFile, tlsConfig.KeyFile,
-			tlsConfig.CAFile, tlsConfig.ServerName, tlsConfig.InsecureSkipVerify)
+		tlsClientConfig, err := thanos_tls.NewClientConfig(logger, userTLSConfig.CertFile, userTLSConfig.KeyFile,
+			userTLSConfig.CAFile, userTLSConfig.ServerName, userTLSConfig.InsecureSkipVerify)
 
 		if err != nil {
 			return nil, err
 		}
 
-		opts.TLSConfig = tlsClientConfig
+		tlsConfig = tlsClientConfig
 	}
 
-	redisClient := redis.NewClient(opts)
-	if reg != nil {
-		reg = prometheus.WrapRegistererWith(prometheus.Labels{"name": name}, reg)
+	var disableCaching = false
```

Comment:
Why not `disableCaching := false`? I thought we used `var` only to declare variables and use their type's zero-value.

Reply:
Also, this is client-side caching, right? Could the variable be named `clientSideCacheEnabled` then? It is more concise with what it really is and avoids the "negative thinking" (which causes double negation when `disableCaching = false`.